grakn-graql-shell/src/main/java/ai/grakn/graql/GraqlShell.java

```diff
@@ -222,10 +208,17 @@ public static boolean runShell(String[] args, String version, String historyFile
         }
 
         Keyspace keyspace = Keyspace.of(cmd.getOptionValue("k", DEFAULT_KEYSPACE));
-        SimpleURI location = Optional.ofNullable(cmd.getOptionValue("r")).map(SimpleURI::new).orElse(Grakn.DEFAULT_URI);
+
+        int defaultGrpcPort = 48555;
+        SimpleURI defaultGrpcUri = new SimpleURI(Grakn.DEFAULT_URI.getHost(), defaultGrpcPort);
+
+        Optional<SimpleURI> location = Optional.ofNullable(cmd.getOptionValue("r")).map(SimpleURI::new);
         String outputFormat = cmd.getOptionValue("o", DEFAULT_OUTPUT_FORMAT);
 
-        if (!client.serverIsRunning(location)) {
+        SimpleURI httpUri = location.orElse(Grakn.DEFAULT_URI);
+        SimpleURI grpcUri = location.orElse(defaultGrpcUri);
```

Comment:
Question: `location` can be something like `localhost:48555` if specified via `-r` so then `httpUri` would become `localhost:48555`. Basically if I specify a uri via `-r` then `httpURI==grpcUri`, is that intended? What is the difference between the two then?

Reply:
One of them is the HTTP port for our REST API (so 4567 by default), the other is for gRPC (so 48555 by default).

I don't know if that behaviour is OK, I'll have to check.

Reply:
Update: it's not OK because we use that URI to check if the server is running or not, which must be over HTTP.

I don't know what to do about that :(